src/components/Drawer.js

```diff
@@ -0,0 +1,217 @@
+        anchor={anchor}
+        open={open}
+        classes={{
+          paper: classes.drawerPaper,
+        }}
+      >
+        <div className={classes.drawerHeader}>
+
+        <Tooltip title="Auto Associate Hydrants" placement="top-start">
+          <IconButton>
+            <CallMerge />
+          </IconButton>
+        </Tooltip>
+
+        <ImportExport
+          importKMLClicked={importKMLClicked}
+          trails={trails}
+          hydrants={hydrants}
+        />
+
+        <Typography variant="title" color="inherit">
+          {drawerHeaderTitle[mode]}
+        </Typography>
+          <IconButton onClick={this.handleDrawerClose}>
+            {theme.direction === 'rtl' ? <ChevronRightIcon /> : <ChevronLeftIcon />}
+          </IconButton>
+
+        </div>
+
+        <TrailList
+          modifyTrail={modifyTrail}
+          canCreate={canCreate}
+          toggleCreate={toggleCreate}
+          trails={trails}
+          trailSelected={trailSelected}
+          mode={mode}
+          hydrants={hydrants}
+          selected={selectedTrail}
+        />
+      </Drawer>
+    );
+
+
+    let before = null;
+    let after = null;
+
+    if (anchor === 'left') {
+      before = drawer;
+    } else {
+      after = drawer;
+    }
+
+    return (
+      <div className={classes.root}>
+        <div className={classes.appFrame}>
+          <AppBar
+            className={classNames(classes.appBar, {
+              [classes.appBarShift]: open,
+              [classes[`appBarShift-${anchor}`]]: open,
+            })}
+          >
+            <Toolbar disableGutters={!open}>
+              <IconButton
+                color="inherit"
+                aria-label="open drawer"
+                onClick={this.handleDrawerOpen}
+                className={classNames(classes.menuButton, open && classes.hide)}
+              >
+                <MenuIcon />
+              </IconButton>
+              <Typography variant="title" color="inherit" noWrap>
+                SnoTrack
+              </Typography>
+              <MapControls
+                mode={mode}
+                changeMode={changeMode}
+              />
+            </Toolbar>
+            <div id="searchLocations"></div>
+          </AppBar>
+          {before}
+          <main
+            className={classNames(classes.content, classes[`content-${anchor}`], {
+              [classes.contentShift]: open,
+              [classes[`contentShift-${anchor}`]]: open,
+            })}
+          >
+            <div className={classes.drawerHeader} />
+
+            <OpenLayersMap
+              mode={mode}
+              canCreate={canCreate}
+              createObject={createObject}
+              modifyTrail={modifyTrail}
+              modifyHydrant={modifyHydrant}
+              trails={trails}
+              hydrants={hydrants}
+              selectedTrail={selectedTrail}
+            />
+
+          </main>
+          {after}
+        </div>
+      </div>
+    );
+  }
+}
```

Comment:
Render methods should be simple logic and then the html. If you need to get html from somewhere else, like with the drawer, put it in an internal `renderDrawer` method you can call or it might mean it should just be it's own component. 

We've erred towards fewer component files in this application so far, but that's actually counter to React best practices, which tend to prefer making lots of smaller, more modular components. Don't be afraid to throw some new files out there.

Reply:
Also, don't call this file/component `Drawer` if you keep it as is, since it actually is everything, and NOT just the drawer. 

It seems from an application standpoint, container should probably be rendering a `drawer` component that is just the drawer stuff (with TrailList or whatever as child components likely), and a `map` or `main` component or something like that to hold the rest. Matches with how things come out on the page, which is nice.